packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -640,10 +641,39 @@ class RenderParagraph extends RenderBox
     );
   }
 
+  bool _systemFontsChangeScheduled = false;
   @override
   void systemFontsDidChange() {
-    super.systemFontsDidChange();
-    _textPainter.markNeedsLayout();
+    final SchedulerPhase phase = SchedulerBinding.instance.schedulerPhase;
+    switch (phase) {
+      case SchedulerPhase.idle:
+      case SchedulerPhase.postFrameCallbacks:
+        if (_systemFontsChangeScheduled) {
+          return;
+        }
+        _systemFontsChangeScheduled = true;
+        SchedulerBinding.instance.scheduleFrameCallback((Duration timeStamp) {
+          if (!_systemFontsChangeScheduled) {
+            return;
+          }
```

Comment:
Do we need this guard? It looks like you only schedule the callback when _systemFontsChangeScheduled is false, and the only way to reset it back to false is in the callback. Is it ever possible that we have a callback scheduled but don't want to run it anymore?

Maybe just turn this into an assert?